### PR TITLE
git-debuggability: allow site-admins configure recorded commands on all repositories

### PR DESCRIPTION
**Pull request by BolajiOlajide**
I have pretty several repos on my local Sourcegraph instance, and I'll like to record commands on all of them instead of adding each of them one at a time to the site config.

## Test plan

* Update your site-config to include the `gitRecorder` configuration. 
* Update the `repos` array to include a wildcard

```json
"gitRecorder": {
    "size": 200,
    "repos": ["*"]
  },
```

* Recorded commands will be stored on all repos.

### CODE REVIEW

File: schema/site.schema.json [hunk 1]
```diff
@@ -1935,7 +1935,7 @@
           "maximum": 10000
         },
         "repos": {
-          "description": "List of repositories whose git operations should be recorded.",
+          "description": "List of repositories whose git operations should be recorded. To record commands on all repositories, simply pass in a wildcard as the only item in the array.",
```
Comment (sashaostrikov) [inline schema/site.schema.json:1938]: I'd phrase it this way to remove possible ambiguity between `*` and `.*`?
```suggestion
          "description": "List of repositories whose git operations should be recorded. To record commands on all repositories, simply pass in an asterisk as the only item in the array.",
```

File: cmd/gitserver/shared/shared.go [hunk 2]
```diff
@@ -642,6 +642,14 @@ func recordCommandsOnRepos(repos []string, ignoredGitCommands []string) wrexec.S
 		ignoredGitCommands = append(ignoredGitCommands, defaultIgnoredGitCommands...)
 	}
 
+	// If repos contains a single "*" element, it means to record commands
+	// for all repositories. In that case, always return true.
+	if len(repos) == 1 && repos[0] == "*" {
```
Comment (sashaostrikov) [inline cmd/gitserver/shared/shared.go:647]: Hmm, do we need to move it to L663? So that we don't skip the ignored commands check?

**Review by sashaostrikov — COMMENTED** [review]
I think we might need to move your check to a slightly different place.

File: cmd/gitserver/shared/shared.go (hunk 2 above)
Comment (sashaostrikov) [reply cmd/gitserver/shared/shared.go:647]: And maybe it's also a good idea to support `*` for commands?

Comment (BolajiOlajide) [reply cmd/gitserver/shared/shared.go]: Yeah, I forgot to push that commit. I just did.
Thanks.

**Review by sashaostrikov — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**